packages/components/data-table-manager/src/data-table-manager.js

```diff
@@ -37,77 +40,202 @@ const DataTableManager = (props) => {
+     */
     secondaryButton: PropTypes.element,
   }),
+  /**
+   * The managed column settings of the table.
+   */
   columnManager: PropTypes.shape({
+    /**
+     * Set this to `false` to show the column settings panel option.
+     *
+     * @@defaultValue@@: true
+     */
     disableColumnManager: PropTypes.bool,
-    visibleColumnKeys: requiredIf(
-      PropTypes.array,
-      (props) =>
-        props.columnManager && !props.columnManager.disableColumnManager
-    ),
-    hideableColumns: requiredIf(
-      PropTypes.array,
-      (props) =>
-        props.columnManager && !props.columnManager.disableColumnManager
+    /**
+     * The keys of the visible columns.
+     */
+    visibleColumnKeys: PropTypes.arrayOf(PropTypes.string.isRequired),
+    /**
+     * The keys of the visible columns.
+     */
+    hideableColumns: PropTypes.arrayOf(
+      PropTypes.shape({
+        key: PropTypes.string.isRequired,
+        label: PropTypes.oneOfType([PropTypes.string, PropTypes.node])
+          .isRequired,
+      })
     ),
+    /**
+     * Set this to `true` to show a search input for the hidden columns panel.
+     */
     areHiddenColumnsSearchable: PropTypes.bool,
-    searchHiddenColumns: requiredIf(
-      PropTypes.func,
-      (props) =>
-        props.columnManager &&
-        !props.columnManager.disableColumnManager &&
-        props.columnManager.areHiddenColumnsSearchable
-    ),
+    /**
+     * A callback function, called when the search input for the hidden columns panel changes.
+     * <br>
+     * Signature: `(event) => void`
+     */
+    searchHiddenColumns: PropTypes.func,
+    /**
+     * Placeholder value of the search input for the hidden columns panel.
+     */
     searchHiddenColumnsPlaceholder: PropTypes.string,
+    /**
+     * A React element to be rendered as the primary button, useful when the column settings work as a form.
+     */
     primaryButton: PropTypes.element,
+    /**
+     * A React element to be rendered as the secondary button, useful when the column settings work as a form.
+     */
     secondaryButton: PropTypes.element,
   }),
-  onSettingsChange: requiredIf(
-    PropTypes.func,
-    (props) =>
-      !props.displaySettings.disableColumnManager ||
-      !props.columnManager.disableDisplaySettings
-  ),
-  /* Top bar */
+  /**
+   * A callback function, called when any of the properties of either display settings or column settings is modified.
+   * <br>
+   * Signature: `(action: string, nextValue: object) => void`
+   */
+  onSettingsChange: PropTypes.func,
+  /**
+   * A React node for rendering additional information within the table manager.
+   */
   topBar: PropTypes.node,
 };
-DataTableManager.defaultProps = {
-  isCondensed: false,
-  itemRenderer: (row, column) => row[column.key],
-  displaySettings: {
-    disableDisplaySettings: true,
-  },
-  columnManager: { disableColumnManager: true },
```

Comment:
Each `displaySettings` or `columnManager` object is in theory optional, so we shouldn't initialize it here.